### PR TITLE
Sidebar

### PR DESCRIPTION
**Pull request by Blistin**

### CODE REVIEW

**Comment by review-ninja** [comment]
<a href="https://app.review.ninja/3See/etect-app/pull/23" target="_blank">ReviewNinja

File: modules/core/client/controllers/groupController.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
+'use strict';
+
+angular.module('core').controller('GroupCtrl', function ($scope, $log) {
```
Comment (Blistin) [inline modules/core/client/controllers/groupController.js]: !fix Can you please comment on what 'log' is? Just comment the code a little better about what is happening here

Comment (Blistin) [inline modules/core/client/controllers/groupController.js]: I know it is the log output being used but someone might not know that on the team

File: modules/core/client/controllers/subjectSetupController.js [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+      "10005",
+      "10006"
+    ];
+  }
+]);
+
+
+
+
+angular.module('core').controller('GroupCtrl', function ($scope, $log) {
```
Comment (Blistin) [inline modules/core/client/controllers/subjectSetupController.js]: !fix We should probably not use two duplicate declarations of GroupCtrl controller. Try to get this into its own file.

File: modules/core/client/controllers/subjectSetupController.js [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+
+
+
+
+
+
+
+
+
+angular.module('core').controller('DatepickerDemoCtrl', function ($scope) {
```
Comment (Blistin) [inline modules/core/client/controllers/subjectSetupController.js]: !fix Move this controller into it's own file, it shouldn't be grouped here if it is being used like a directive.

File: modules/core/client/views/enroll.client.view.html [hunk 4]
```diff
@@ -1,4 +1,5 @@
-<section data-ng-controller="EnrollController">
+<section data-ng-controller="EnrollController" id="content">
```
Comment (Blistin) [inline modules/core/client/views/enroll.client.view.html:1]: !fix Instead of declaring this outside the content like this, it would be better to declare this inside of the side-bar directive!

Like so:

<side-bar></side-bar>

(Inside of side-bar.html)
<section data-ng-controller="EnrollController" id="content">
//side-bar.html goes here
</section>

Comment (Blistin) [inline modules/core/client/views/enroll.client.view.html:1]: !fix Ah, I see. You have a separate sidebar for each view. You need to find a way to get the sidebar into the index.html file in core>server, kinda like how the header is done to where it is the same one for each view and the views are changing but the header is static.

File: modules/core/client/views/sidebar.client.view.html [hunk 5]
```diff
@@ -0,0 +1,40 @@
+<section id="sidebar">
+    <br>
```
Comment (Blistin) [inline modules/core/client/views/sidebar.client.view.html]: !fix Using br to move stuff down is bad practice, use a custom margin class to push down content if you must. !fix

File: modules/core/client/views/sidebar.client.view.html [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+    <br>
+    <br>
+    <div ng-controller="sidebarCtrl">
```
Comment (Blistin) [inline modules/core/client/views/sidebar.client.view.html]: !fix Should be declared in the section tag at the top of the html content, where id="sidebar" is currently defined

File: modules/core/client/views/subject_setup.client.view.html [hunk 7]
```diff
@@ -0,0 +1,84 @@
...
+                    <option ng-repeat= "reader in readers" value="{{reader}}">{{reader}}</option>
+                  </select>
+            </td>
+        </tr>  
+        <tr>
+            <td>Start Date</td>
+            <td>
+
+                
+                <div ng-controller="DatepickerDemoCtrl">
```
Comment (Blistin) [inline modules/core/client/views/subject_setup.client.view.html]: !fix Instead of having the controller declared for both of these sections separately just wrap them in a div around the scope of both of these declarations and use that to define the controller. That way it will be able to be scoped there with one declaration!

File: modules/core/client/views/subject_setup.client.view.html [hunk 8]
```diff
@@ -0,0 +1,84 @@
...
+                      </span>
+                    </p>
+                </div>
+
+            </td>
+        </tr>
+        <tr>
+            <td>End Date</td>
+            <td>
+                <div ng-controller="DatepickerDemoCtrl">
```
Comment (Blistin) [inline modules/core/client/views/subject_setup.client.view.html:66]: !fix Same as other datepickerdemoctrl declaration above

File: modules/core/server/views/layout.server.view.html [hunk 9]
```diff
@@ -38,18 +38,21 @@
...
+    </section>
+  -->
+    <section id="banner">
       {% block banner %}{% endblock %}
     </section>
-  </section>
-  <section class="content">
-    <section class="container">
-      {% block content %}{% endblock %}
+    <section>
```
Comment (Blistin) [inline modules/core/server/views/layout.server.view.html]: !fix Why the change of this? This is going to remove the container class from every view we use inside of the block-container setup. Probably need to reset this.

File: modules/core/server/views/layout.server.view.html [hunk 10]
```diff
@@ -38,18 +38,21 @@
 </head>
 
 <body class="ng-cloak">
-  <header data-ng-include="'/modules/core/views/header.client.view.html'" class="navbar navbar-fixed-top navbar-inverse"></header>
-  <!--Found in index.server.view.html-->
-  <section class="banner">
-    <section class="container">
+  <div class="container">
```
Comment (Blistin) [inline modules/core/server/views/layout.server.view.html]: !fix I see what you are going for by killing two birds here with wrapping all the views in a container, but it might be best if we just let the containers be declared above the views they will be needed for here. We might not WANT the same padding on the sidebar that we will get on the page content views or the headers, so probably best to keep this separate for now.

File: package.json [hunk 11]
```diff
@@ -40,7 +40,7 @@
     "lodash": "^3.10.0",
     "method-override": "^2.3.3",
     "mocha": "~1.20.0",
-    "mongoose": "^4.0.6",
+    "mongoose": "^4.1.11",
```
Comment (Blistin) [inline package.json:43]: !fix  You need to be careful about pushing up changes to package.json. I know this is to fix the whole mongodb auth issue but make sure you let everyone know that you are going to do this because if you pushed up a different change to a working package it could break more things than it fixes!

File: modules/core/client/views/enroll.client.view.html (hunk 4 above)
Comment (naugle92) [inline modules/core/client/views/enroll.client.view.html:1]: !fixed It's the same side bar, it's just being called through a directive. I tried putting it into the index file, but I couldn't find a way for it to not appear on the login screen.

File: modules/core/client/controllers/groupController.js (hunk 1 above)
Comment (naugle92) [inline modules/core/client/controllers/groupController.js]: !fixed deleted $log. it was not necessary.

File: modules/core/client/views/enroll.client.view.html (hunk 4 above)
Comment (naugle92) [inline modules/core/client/views/enroll.client.view.html:1]: !fixed I couldn't put the sidebar into the index file without having it in the login screen. I did it this way in order to prevent that

File: package.json (hunk 11 above)
Comment (naugle92) [inline package.json:43]: !fixed everyone notified

File: modules/core/client/views/subject_setup.client.view.html (hunk 8 above)
Comment (naugle92) [inline modules/core/client/views/subject_setup.client.view.html:66]: !fixed placed in separate file

File: modules/core/client/controllers/subjectSetupController.js (hunk 2 above)
Comment (naugle92) [inline modules/core/client/controllers/subjectSetupController.js]: !fixed groupctrl was placed in its own file

File: modules/core/client/controllers/subjectSetupController.js (hunk 3 above)
Comment (naugle92) [inline modules/core/client/controllers/subjectSetupController.js]: !fixed placed into its own file

File: modules/core/client/views/sidebar.client.view.html (hunk 5 above)
Comment (naugle92) [inline modules/core/client/views/sidebar.client.view.html]: !fixed br removed, padding added

File: modules/core/client/views/sidebar.client.view.html (hunk 6 above)
Comment (naugle92) [inline modules/core/client/views/sidebar.client.view.html]: !fixed controller declaration now in section tag

File: modules/core/client/views/subject_setup.client.view.html (hunk 7 above)
Comment (naugle92) [inline modules/core/client/views/subject_setup.client.view.html]: !fixed controller declared once in table tag, removed other two instances

File: modules/core/server/views/layout.server.view.html (hunk 9 above)
Comment (naugle92) [inline modules/core/server/views/layout.server.view.html]: !fixed returned to original

File: modules/core/server/views/layout.server.view.html (hunk 10 above)
Comment (naugle92) [inline modules/core/server/views/layout.server.view.html]: !fixed returned to original

File: modules/core/client/controllers/groupController.js (hunk 1 above)
Comment (Blistin) [inline modules/core/client/controllers/groupController.js]: !fix remove the $log from the function since it isn't being used

Comment (naugle92) [inline modules/core/client/controllers/groupController.js]: !fixed removed
